R/featEng-tails.R

```diff
@@ -0,0 +1,166 @@
+          quantile_cutoff_upper <- quantile(uniqueid_df[, 3], 1-alpha)
+
+          # find values that fall beneath and above quantile cutoffs
+          values_lower <- uniqueid_df[, 3][which(uniqueid_df[, 3] < quantile_cutoff_lower)]
+          values_upper <- uniqueid_df[, 3][which(uniqueid_df[, 3] > quantile_cutoff_upper)]
+
+          # finding averages
+          mean_lower <- mean(values_lower)
+          mean_upper <- mean(values_upper)
+
+          # appending all values into a vector
+          clust_vec <- c(clust_vec, i)
+          PID_vec <- c(PID_vec, ID)
+          lower_mean_vec <- c(lower_mean_vec, mean_lower)
+          upper_mean_vec <- c(upper_mean_vec, mean_upper)
+          lower_quantile_vec <- c(lower_quantile_vec, quantile_cutoff_lower)
+          upper_quantile_vec <- c(upper_quantile_vec, quantile_cutoff_upper)
+
+          }
+
+        }
+
+      # putting all results into a dataframe
+      result <- data.frame(PID = PID_vec,
+                       Cluster = clust_vec,
+                       LowerMean = lower_mean_vec,
+                       UpperMean = upper_mean_vec,
+                       LowerQuantile = lower_quantile_vec,
+                       UpperQuantile = upper_quantile_vec)
+
+  }
+
+    else {
+
+      new_df <- voxel_df[, c(2:6)]
+
+      PID_vec <- c()
+      lower_mean_vec <- c()
+      upper_mean_vec <- c()
+      lower_quantile_vec <- c()
+      upper_quantile_vec <- c()
+
+      for (ID in unique(new_df[, 1])) {
+        # filter the dataframe so that we are only considering individual PIDs
+        uniqueid_df <- new_df %>% filter(pid == ID)
+
+        # find quantile cutoffs
+        quantile_cutoff_lower <- quantile(uniqueid_df[, 5], alpha)
+        quantile_cutoff_upper <- quantile(uniqueid_df[, 5], 1-alpha)
+
+        # find values that fall beneath and above quantile cutoffs
+        values_lower <- uniqueid_df[, 5][which(uniqueid_df[, 5] < quantile_cutoff_lower)]
+        values_upper <- uniqueid_df[, 5][which(uniqueid_df[, 5] > quantile_cutoff_upper)]
+
+        # finding averages
+        mean_lower <- mean(values_lower)
+        mean_upper <- mean(values_upper)
+
+        # appending all values into a vector
+        PID_vec <- c(PID_vec, ID)
+        lower_mean_vec <- c(lower_mean_vec, mean_lower)
+        upper_mean_vec <- c(upper_mean_vec, mean_upper)
+        lower_quantile_vec <- c(lower_quantile_vec, quantile_cutoff_lower)
+        upper_quantile_vec <- c(upper_quantile_vec, quantile_cutoff_upper)
+
+      }
+
+      # putting all results into dataframe
+      result <- data.frame(PID = PID_vec,
+                          LowerMean = lower_mean_vec,
+                          UpperMean = upper_mean_vec,
+                          LowerQuantile = lower_quantile_vec,
+                          UpperQuantile = upper_quantile_vec)
+      }
+
+  }
+
+  else {
+
+    stop("alpha cannot be 0 or 1")
+
+  }
+
+return(result)
```

Comment:
Indent
We want to adhere to the tidyverse style guide: https://style.tidyverse.org/